ARIES-2165: Build proxy on Java 11, 17 and 21

alien11689 opened the pull request:
- [x] Remove `sun.reflect.ReflectionFactory` usage

alien11689 committed:
ARIES-2165: Update parent pom and build proxy on Java 11, 17 and 21 adding objensis

alien11689 commented on proxy/proxy-impl/src/test/java/org/apache/aries/blueprint/proxy/ProxySubclassGeneratorTest.java:
```diff
@@ -173,6 +174,7 @@ public void testFinalClass() throws Exception
    * Test a private constructor
    */
   @Test
+  @Ignore
```
I haven't found good replacement for sun.reflect.* to make private constructor accessible

alien11689 replied:
objenesis seems to be good replacement

jbonofre commented:
I'm testing it fully, it looks good so far.

I just wonder if we strictly have to keep JDK8. Thoughts ?

alien11689 commented:
It's probably hard to measure but I think we have still a lot of Java 8 users. Additionally, the biggest subprojects - blueprint and subsystem - are not buildable with Java > 8.

Indeed keeping support for Java LTS 8 to 21 is quite hard and now there is also 25, so limiting the number of supported version would make the maintenance easier.

What Java version would you recommend @jbonofre ?

jbonofre commented:
@alien11689 I would say that JDK 17 and 21 are good enough. I will resume work on this one.

alien11689 commented:
thank you @jbonofre 
I quick rebase should be enough on this PR, there should be not a lot of conflicts

alien11689 committed:
ARIES-2165: Test proxy on java 25

alien11689 commented:
Hi @jbonofre 
I have rebased this branch and added support also for java 25
This PR is nearly one year old - if you don't mind, I'll merge it on Sunday
